### PR TITLE
Extends selectLocalImage and fileChanged methods

### PR DESCRIPTION
**Pull request by Ckpuna4**

### CODE REVIEW

**Comment by MohammedAlabd** [comment]
looks like a good PR, why is it still hanging there?

**Comment by Ckpuna4** [comment]
Sure

When I started updating the PATCH version from 1.2.23 to
1.2.26, I was surprised to find backward compatibility-breaking changes (deleted/renamed methods, public methods that have become private)

Before patch changes I used `imageDrop` and its public methods
`const images = imageDrop.getImageFiles(event.target.files);`
`imageDrop.handleNewImageFiles(images);`

after patch changes I have to use `imageCompress` and `imageCompress.fileChanged` and I had to modify the method `fileChanged` so that it could work with files from the outside

As for the `selectLocalImage` method
I thought it would be nice for a synchronous method that internally uses an asynchronous operation to add the possibility of notification after performing an asynchronous operation

File: src/quill.imageCompressor.ts [hunk 1]
```diff
@@ -68,8 +68,8 @@ class imageCompressor {
     });
   }
 
-  async fileChanged() {
-    const files = this.fileHolder?.files;
+  async fileChanged(files?: File[]) {
```
Comment (benwinding) [inline src/quill.imageCompressor.ts:71]: @Ckpuna4 why is this property optional? Are you using this externally?

Comment (Ckpuna4) [reply src/quill.imageCompressor.ts:71]: >why is this property optional?

To preserve backwards compatibility, if someone is already using this method in their project

>Are you using this externally?

After upgrading to the latest version — yes

**Comment by benwinding** [comment]
Well LGTM, merged and now available in `quill-image-compress@1.2.27`
Thank you for the contribution and providing the context of the changes 👍

**Pull request merged**